provider/pkg/provider/provider.go

```diff
@@ -733,14 +734,7 @@ func (p *cfnProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*
 			return nil, errors.Errorf("Resource type %s not found", resourceToken)
 		}
 
-		if autoNamingSpec := spec.AutoNamingSpec; autoNamingSpec != nil {
-			// Auto-name fields if not already specified
-			val, err := getDefaultName(req.RandomSeed, urn, autoNamingSpec, olds, newInputs)
-			if err != nil {
-				return nil, err
-			}
-			newInputs[resource.PropertyKey(autoNamingSpec.SdkName)] = val
-		}
+		autonaming.ApplyAutoNaming(spec.AutoNamingSpec, urn, req.RandomSeed, olds, newInputs)
```

Comment:
I think there's a behavior change here: the old code returned an error if there was an error from `getDefaultName` but it looks like you're dropping it here.

Reply:
Great spot! Thank you 👍